src/EnergyPlus/MixedAir.cc

```diff
@@ -3946,8 +3957,8 @@ Real64 VentilationMechanicalProps::CalcMechVentController(EnergyPlusData &state,
                                         // Calculate zone maximum target CO2 concentration in PPM
                                         if (this->SystemOAMethod == DataSizing::SysOAMethod::ProportionalControlDesOcc) {
                                             // Accumulate CO2 generation from people at design occupancy and current activity level
-                                            for (int PeopleNum = 1; PeopleNum <= state.dataHeatBal->TotPeople; ++PeopleNum) {
-                                                if (state.dataHeatBal->People(PeopleNum).ZonePtr != ZoneNum) continue;
+                                            Real64 CO2PeopleGeneration = 0.0;
+                                            for (int const PeopleNum : thisMechVentZone.peopleIndexes) {
```

Comment:
Loop through only the applicable people instances, not all of them. Also reset sum to zero for each zone (previously this kept accumulating across zones).